Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -752,44 +947,33 @@ namespace OpenInfraPlatform {
 						const uint32_t v1 = mergedIndices[i1];
 						const uint32_t v2 = mergedIndices[i2];
 
-						if(faceLoopReversed)
+						if (faceLoopReversed)
 							polygon->addFace(v0, v2, v1);
 						else
 							polygon->addFace(v0, v1, v2);
 					}
 				}
 
-				bool convertIfcBSplineSurface(const std::shared_ptr<typename IfcEntityTypesT::IfcBoundedSurface>& boundedSurface,
-					const carve::math::Matrix& pos,
-					std::shared_ptr<carve::input::PolylineSetData> polyline_data)
-				{
-					// all versions < IFC4 do not support IfcBSplineSurface entities
-					return false;
-				}
-
-				void convertIfcAdvancedFaceList(const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcFace>>& faces,
-					const carve::math::Matrix& pos,
-					std::shared_ptr<ItemData> item_data,
-					std::stringstream& strs_err)
-				{
-					// nothing to do here as it is not supported by older IFC versions (< IFC4)
-				}
+				/*! \brief Converts \c IfcCartesianPoint to a 2D vector.
+				\param	faceVertices2D to be triangulated using carve
+				\param	faceVertices3D to be added to the triangulation
+				*/
 
-				void convertIfcCartesianPoint2DVector(const std::vector<std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D,
+				void convertIfcCartesianPoint2DVector(const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D,
 					std::vector<std::vector<carve::geom::vector<3>>>& loop2D) const

```

Comment:
Please convert return `points2D` from the function instead of passing it as out param.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/5078d3d31fe8ba518748be032727b0e685a206b3